sretoolbox/utils/multiprocess.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2021 Red Hat
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Multiprocessing abstractions.
+"""
+
+import multiprocessing
+from multiprocessing import Pool
+
+
+def run(func, iterable, process_pool_size,
+        return_exceptions=False, **kwargs):
+    """run_processes executes a function for each item in the input iterable.
+    execution will be done in a processpool according to the input
+    process_pool_size.  kwargs are passed to the input function
+    (optional). If return_exceptions is true, any exceptions that may
+    have happened in each thread are returned in the return value,
+    allowing the caller to get as much work done as possible.
+    """
+    mpm = multiprocessing.Manager()
+
+    process_list = [(func, [i], kwargs, mpm.dict()) for i in iterable]
+    with Pool(process_pool_size) as pool:
+        pool.map(_process_task, process_list)
+
+    results = []
+    for process_item in process_list:
+        if "success" in process_item[3]:
+            results.append(process_item[3]["success"])
+        elif "exception" in process_item[3] and return_exceptions:
+            results.append(process_item[3]["exception"])
+        elif "exception" in process_item[3]:
+            raise process_item[3]["exception"]
+    return results
+
+
+def _process_task(spec):
+    func, args, kwargs, result = spec
+    try:
+        result["success"] = func(*args, **kwargs)
+    # pylint: disable=broad-except
+    except Exception as details:
+        result["exception"] = details
+    except SystemExit as details:
```

Comment:
Shouldn't this be caught before `Exception`? I.e., doesn't `Exception` being declared first, swallow this one?

Reply:
Or maybe a comment here why this is needed in the first place?

Reply:
Exception and SystemExit live next to each other in the python exception hierarchy
https://docs.python.org/3/library/exceptions.html#exception-hierarchy